### ISSUE TITLE
[Enhancement] Autolinking bot commands

### ISSUE THREAD

**Issue by Mindstormer619** [DrKLO/Telegram#1190]
Currently any word with a `/` before it is autosent on tap, like so:

![spam of botwords](https://cloud.githubusercontent.com/assets/441849/8350182/d22e6dea-1b40-11e5-85ef-5ce1067b9f70.png)

This has several problems that make it a really bad idea to put into the client:

1. Many bots, like `@alertbot` require commands WITH parameters, like `/alert <time> text`. This autosending makes it send without any related command options, making the bot do weird stuff, like this, which will never execute:  
![useless command](https://cloud.githubusercontent.com/assets/441849/8350206/f9f1daec-1b40-11e5-965b-c5ee754ebc85.png)  
2. People can spam the command in group chats by tapping on them, which is wasteful and might even crash an active bot. Example image above (Note that Telegram has popular public groups after introduction of invite links).
3. Messages that have words preceded by `/` are linked as bot commands even when they are clearly NOT bot commands. For example, consider me telling where to find a program on a Linux path:  
![image](https://cloud.githubusercontent.com/assets/441849/8350230/26302140-1b41-11e5-99ff-2a851636c4b0.png)  
See what I mean? It is linking something when it is clearly not meant to.

## Solutions

1. Do not autosend. Instead, please paste it to the text box so we can append new text to it. This is similar to the behavior of @-mentions, which do not send the text immediately but instead append them to the text in the box. This should also be the case for the command autocomplete here:
![image](https://cloud.githubusercontent.com/assets/441849/8350164/a14a9fbe-1b40-11e5-8da4-4a886163a7ef.png)  
Currently it sends the text immediately instead of autocomplete so it is impossible to add parameters without manually typing. For sending commands without parameters, we can always hit the send key on the keyboard. We're not that lazy.
2. Do not autolink commands unless they are of the form `<space>/text<space>`. This will prevent stuff like `/bin` from `/bin/ls` and other related text from getting linked.

This is not a minor problem. I have set it as an "enhancement" instead of a "bug" because this was probably your intended behavior (though I don't see why: this has absolutely no advantages). This is pretty much an anti-design making it harder to use Telegram instead of easier.

Other than that, thanks for the updates and new features!

**Comment by topkecleon** [comment]
I have also experienced this issue, and I share the concern.

**Comment by neongreen** [comment]
> Do not autolink commands unless they are of the form <space>/text<space>. This will prevent stuff like /bin from /bin/ls and other related text from getting linked.

Please, do not autolink commands *at all* if they aren't in a conversation with a bot. I'm fond of writing “/exhales”, “/doesn't understand”, “/smiles” to refer to myself in chat (like “/me smiles” in Skype) and it's slightly annoying that e.g. “doesn” in “/doesn't understand” gets autolinked. Telegram for Linux gets it right, for instance.

**Comment by Mindstormer619** [comment]
@neongreen :+1: This is actually the right thing to do, but baby steps :grin:

**Comment by ollpu** [comment]
:+1: Very annoying to use commands with parameters.

**Comment by PrathameshBhosale** [comment]
It causes too much spam in the group both unknowingly bymistake or also exploited by spammers to spam in a public group.

**Comment by samuelv89** [comment]
I've had problems with this issue too, at least it should ask the user before sending it. But best if not send at all, just paste it to the text box.

**Comment by Mindstormer619** [comment]
Yeah, that's the solution I gave. Pasting into the text box will allow us to append params and send at our own leisure. Now to wait for the devs...

**Comment by nadam** [comment]
The automatic send is convenient for interfaces such as @BotFather, but I think the app could be clever about that and only auto-send when tapping a command inside a message from a bot (one that is using the new bot API) and only paste when tapping in other messages.

**Comment by ollpu** [comment]
> The automatic send is convenient for interfaces such as @BotFather

**Comment by ollpu** [comment]
> The automatic send is convenient for interfaces such as @BotFather
This could be subject to configuration as well, maybe if @BotFather had a setting for whether to auto-send when tapping a command or not.

**Comment by ShahabJafri** [comment]
Kindly look into this as now this "feature" is being exploited a lot in groups.

**Comment by ADTC** [comment]
Please remove autosend completely. It's very annoying and not in line with the flexibility of bots accepting parameters for their commands. All commands must always paste into the message box.

If the user doesn't have any parameters to add the user can just hit send to send it to the bot. I'm sure it's not an inconvenience to do so. But it's a real inconvenience to have to manually type the full command if we need to add parameters, instead of conveniently pasting from the menu and adding them. Even the autocomplete help is useless.

**Comment by ADTC** [comment]
I had opened #1236 asking for this same thing (before I found this issue). 3 hours ago, @DrKLO closed it with no explanation.

**Comment by Mindstormer619** [comment]
Just saw that. He's being an ass. Don't know why. This is not even supposed to be difficult to implement. It is supposed to use the same idea as @-mentions, which already exist. It's not like it's something "extra", more like something which should have been corrected a long time ago. Call it a "design bug", if you will, though if I had marked it as a bug instead of as an enhancement, he would most likely have closed it immediately stating _"not a bug"_.

**Comment by Mindstormer619** [comment]
Telegram as a messenger is great. It's random stupidity on the various clients that irritates me.
